mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_id_token.go

```diff
@@ -0,0 +1,251 @@
+				},
+			},
+			"include_email": schema.BoolAttribute{
+				Optional: true,
+				Computed: true,
+			},
+			"id_token": schema.StringAttribute{
+				Computed:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func (p *googleEphemeralServiceAccountIdToken) Configure(ctx context.Context, req ephemeral.ConfigureRequest, resp *ephemeral.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	pd, ok := req.ProviderData.(*fwtransport.FrameworkProviderConfig)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *fwtransport.FrameworkProviderConfig, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	// Required for accessing userAgent and passing as an argument into a util function
+	p.providerConfig = pd
+}
+
+func (p *googleEphemeralServiceAccountIdToken) Open(ctx context.Context, req ephemeral.OpenRequest, resp *ephemeral.OpenResponse) {
+	var data ephemeralServiceAccountIdTokenModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	targetAudience := data.TargetAudience.ValueString()
+	creds := fwtransport.GetCredentials(ctx, fwmodels.ProviderModel{}, false, &resp.Diagnostics)
+
+	targetServiceAccount := data.TargetServiceAccount.ValueString()
+	// If a target service account is provided, use the API to generate the idToken
+	if targetServiceAccount != "" {
+		service := p.providerConfig.NewIamCredentialsClient(p.providerConfig.UserAgent)
+		name := fmt.Sprintf("projects/-/serviceAccounts/%s", targetServiceAccount)
+		DelegatesSetValue, _ := data.Delegates.ToSetValue(ctx)
+		tokenRequest := &iamcredentials.GenerateIdTokenRequest{
+			Audience:     targetAudience,
+			IncludeEmail: data.IncludeEmail.ValueBool(),
+			Delegates:    StringSet(DelegatesSetValue),
+		}
+		at, err := service.Projects.ServiceAccounts.GenerateIdToken(name, tokenRequest).Do()
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error calling iamcredentials.GenerateIdToken",
+				err.Error(),
+			)
+			return
+		}
+
+		data.IdToken = types.StringValue(at.Token)
+		resp.Diagnostics.Append(resp.Result.Set(ctx, data)...)
+		return
+	}
+
+	// If no target service account, use the default credentials
+	ctx = context.Background()
+	co := []option.ClientOption{}
+	if creds.JSON != nil {
+		co = append(co, idtoken.WithCredentialsJSON(creds.JSON))
+	}
+
+	idTokenSource, err := idtoken.NewTokenSource(ctx, targetAudience, co...)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to retrieve TokenSource",
+			err.Error(),
+		)
+		return
+	}
+	idToken, err := idTokenSource.Token()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to retrieve Token",
+			err.Error(),
+		)
+		return
+	}
+
+	data.IdToken = types.StringValue(idToken.AccessToken)
+	resp.Diagnostics.Append(resp.Result.Set(ctx, data)...)
+}
+
+func StringSet(d basetypes.SetValue) []string {
```

Comment:
this needs to be removed once https://github.com/GoogleCloudPlatform/magic-modules/pull/12140 is merged

Reply:
removed in recent rebase / commit